Assets/MRTK/Providers/XRSDK/XRSDKDeviceManager.cs

```diff
@@ -59,31 +63,36 @@ public override void Update()
                     return;
                 }
 
-                // Need to make it so we get devices that are either controllers or handtracking, the old commented code only got devices that met both requirements
-                InputDevices.GetDevicesWithCharacteristics(InputDeviceCharacteristics.Controller, inputDevices);
-                foreach (InputDevice device in inputDevices)
+                inputDevices.Clear();
+                for(int i = 0; i < DesiredCharacteristics.Length; i++)
```

Comment:
One thing we might want to do here is make sure that devices that are returned from previous `InputDevices.GetDevicesWithCharacteristics` calls aren't re-updated or re-detected. I think the re-detected case is okay with the existing logic, but it feels like we'd be updating the same controller multiple times potentially.

Reply:
If GetDevicesWithCharacteristics works like I believe it works (only does an exact match) I don't think we should get the redirected/reupdated case